### PR TITLE
Move DEFAULT_DISPLAY_NAME from app to entity

### PR DESCRIPTION
**Pull request by aledsage**

### CODE REVIEW

**Comment by aledsage** [comment]
Build failure looks like it's caused by infrastructure problems:

```
Modules changed, recalculating dependency graph
Build timed out (after 60 minutes). Marking the build as aborted.
Build was aborted
Putting comment on the pull request
Finished: ABORTED
```

I've done a `git commit --amend && push -f` to try to trigger another jenkins build.

**Comment by neykov** [comment]
Looks like jenkins gave up on this PR entirely. Merging.

File: camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/EntityNameYamlTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                "    myconf: myval",
+                "    defaultDisplayName:",
+                "      $brooklyn:formatString:",
+                "      - \"PREFIX%sSUFFIX\"",
+                "      - $brooklyn:config(\"myconf\")");
+        deployAndAssertDisplayName(yaml, "PREFIXmyvalSUFFIX");
+    }
+
+    @Test
+    public void testBrooklynConfig() throws Exception {
```
Comment (geomacy) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/EntityNameYamlTest.java]: I know it's a bit late commenting after a merge, but what's different about this test than `testExplicitDisplayName` above?  Did you mean to update this to refer to Brooklyn property config in some way?

Comment (aledsage) [inline camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/EntityNameYamlTest.java]: Good point @geomacy - I'll delete this duplicate test.

**Comment by sjcorbett** [comment]
Compilation of Brooklyn core fails after pulling this:
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/sam/code/brooklyn.io/brooklyn/brooklyn-server/core/src/test/java/org/apache/brooklyn/core/entity/EntityNameTest.java:[65,40] error: incompatible types: Entity cannot be converted to TestEntity
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 28.458 s
[INFO] Finished at: 2016-06-15T16:21:26+01:00
[INFO] Final Memory: 26M/686M
[INFO] ------------------------------------------------------------------------
```
Line in question:
```java
    public void testDefaultDisplayNameUsesDeferredSupplier() {
        TestEntity entity = app.addChild(EntitySpec.create(TestEntity.class)
```

Tried both Oracle JDK 1.7 and 1.8.

**Comment by neykov** [comment]
Thanks for pointing out @sjcorbett. Fixed in https://github.com/apache/brooklyn-server/commit/3bb69b6080fbde17279753ba562f4a5008bbd0f6.
The Eclipse compiler is too smart sometimes, especially around generics and losing type info, able to figure out things which leave the java compiler wondering.

**Pull request merged**